src/DataProtection/DataProtection/src/TypeForwardingActivator.cs

```diff
@@ -64,6 +62,26 @@ internal object CreateInstance(Type expectedBaseType, string originalTypeName, o
         }
 
         protected string RemoveVersionFromAssemblyName(string forwardedTypeName)
-            => _versionPattern.Replace(forwardedTypeName, "");
+        {
+            // Type, Assembly, Version={Version}, Culture={Culture}, PublicKeyToken={Token}
+
+            var versionStartIndex = forwardedTypeName.IndexOf("Version=");
+
+            if (versionStartIndex == -1)
+            {
+                // No version?
+                return forwardedTypeName;
+            }
+
+            var versionEndIndex = forwardedTypeName.IndexOf(',', versionStartIndex);
+
+            if (versionEndIndex == -1)
+            {
+                // No end index?
+                return forwardedTypeName;
```

Comment:
Should this be:
```C#
return forwardedTypeName.Substring(0, versionStartIndex);
```
?  Otherwise it could contain a `Version=something` but nothing after it?